aiida/cmdline/commands/cmd_node.py

```diff
@@ -61,6 +64,87 @@ def repo_ls(node, relative_path, color):
         echo.echo_critical(exception)
 
 
+@verdi_node_repo.command('dump')
+@click.option(
+    '-f',
+    '--force',
+    help='Overwrite existing directories and files without prompting for confirmation.',
+    flag_value=True
+)
+@arguments.NODE()
+@click.argument(
+    'output_directory',
+    type=click.Path(exists=True, file_okay=False, writable=True),
+    required=True,
+)
+@with_dbenv()
+def repo_dump(node, output_directory, force):
+    """Copy the repository files of a node to an output directory."""
+    from aiida.orm.utils.repository import FileType
+
+    output_directory = pathlib.Path(output_directory)
+
+    output_directory.mkdir(exist_ok=True)
+
+    def _copy_tree(key, output_dir):  # pylint: disable=too-many-branches
+        for file in node.list_objects(key=key):
+            if not key:
+                file_path = file.name
+            else:
+                file_path = key + '/' + file.name
+            if file.type == FileType.DIRECTORY:
+                new_out_dir = output_dir / file.name
+                if new_out_dir.exists():
+                    if force:
+                        prompt_res = 'o'
+                    else:
+                        prompt_res = click.prompt(
+                            "Directory '{}' exists. Do you wish to abort [a], "
+                            'skip this directory [s], merge directory contents [m], '
+                            'or overwrite the directory (delete current contents) [o]?'.format(new_out_dir),
+                            default='a',
+                            type=click.Choice(['a', 's', 'm', 'o'])
+                        )
+                    if prompt_res == 'a':
+                        raise click.Abort
+                    elif prompt_res == 's':
+                        continue
+                    elif prompt_res == 'o':
+                        shutil.rmtree(new_out_dir)
+                        new_out_dir.mkdir()
+                    else:
+                        assert prompt_res == 'm'
+                else:
+                    new_out_dir.mkdir()
+
+                _copy_tree(key=file_path, output_dir=new_out_dir)
+            else:
+                out_file_path = (output_dir / file.name)
+                if out_file_path.exists():
+                    if force:
+                        prompt_res = 'o'
+                    else:
+                        prompt_res = click.prompt(
+                            'File {} exists. Do you wish to abort [a], skip this file [s], or overwrite the file [o]?'.
```

Comment:
This line is rather long, perhaps better like this?
```suggestion
                            "File '{}' exists. Abort [a], skip [s], or replace [r]?".
```